src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -0,0 +1,272 @@
+  local coinObjects = getObjectsWithTag("ImporterToken")
+  if #coinObjects == 0 then
+    broadcastToAll("Could not find importer token", Color.Red)
+  elseif #coinObjects > 1 then
+    broadcastToAll("More than 1 importer token found. Please delete all but 1 importer token", Color.Yellow)    
+  else
+    coin = coinObjects[1]
+    local importData = JSON.decode(coin.getGMNotes())
+    campaignBoxGUID = importData["box"]
+    local campaignBox = getObjectFromGUID(campaignBoxGUID)
+    if campaignBox.type == "Generic" then
+      campaignBox.call("buttonClick_download")
+    end
+    Wait.condition(
+      function()
+        if #campaignBox.getObjects() > 0 then
+          placeCampaignFromToken(importData)
+        else
+          createCampaignFromToken(importData)
+        end
+      end,
+      function()
+        local obj = getObjectFromGUID(campaignBoxGUID)
+        if  obj == nil then 
+          return false 
+        else
+          return obj.type == "Bag" and obj.getLuaScript() ~= ""
+        end
+      end,
+      2,
+      function() broadcastToAll("Error loading campaign box") end
+    ) 
+  end
+end
+
+-- After box has been downloaded, places content on table
+function placeCampaignFromToken(importData)
+  getObjectFromGUID(campaignBoxGUID).call("buttonClick_place")
+  Wait.condition(
+    function() createCampaignFromToken(importData) end,
+    function() return findCampaignLog() ~= nil end,
+    2,
+    function() broadcastToAll("Error placing campaign box") end
+  )
+end
+
+-- After content is placed on table, conducts all the other import operations
+function createCampaignFromToken(importData)
+  findCampaignLog().destruct()
+  --create campaign log
+  spawnObjectData({data = importData["log"]}) 
+  --create chaos bag
+  chaosBagApi.setChaosBagState(importData["bag"])
+  --populate trauma values
+  if importData[4] then
+    updateCounters(importData["trauma"])
+  end
+  --populate ArkhamDB deck IDs
+  if importData[5] then
+    deckImporterApi.setUiState(importData["decks"])
+  end
+  --set investigator count
+  playAreaAPI.setInvestigatorCount(importData["clueCount"])
+  --set campaign guide page
+  if findCampaignGuide() then
+    Wait.condition(
+      function()
+        --log("Campaign Guide import successful!")
+      end,
+      function()
+        findCampaignGuide().Book.setPage(importData["guide"])
+        return findCampaignGuide().Book.getPage() == importData["guide"]
+      end,
+      1,
+      function()
+        --log("Campaign Guide import failed!")
+      end
+    )
+  end
+  Wait.time(
+    function() optionPanelApi.loadSettings(importData["options"]) end,
+    0.5
+  )
+  broadcastToAll("Campaign successfully imported!", Color.Green)
+end
+
+
+-- Creates a campaign token with save data encoded into GM Notes based on the current state of the table
+function createCampaignToken(_, _, _)
+  local campaignBoxGUID = ""
+  -- find active campaign
+  for _, obj in ipairs(getObjectsWithTag("CampaignBox")) do
```

Comment:
@Entrox-Licher Can you explain how you solved this? I don't really understand how the `campaignBoxGUID` works 😅

Reply:
I'd probably still check to see if you hit two, and abort the process with a message if you do.  If someone just cleans up the old campaign you could easily have two empty boxes. It should be straightforward to get the names and print them

Reply:
(Lines 182-196)
Basically, it will iterate through all the objects that have the CampaignBox tag.  This will return all the campaign boxes, both the downloaded and not.  On each one, it will check to make sure it's downloaded (i.e. is now a Bag) and that it's placed on the table (i.e. empty Bag).  If both of those things are true, it will update the `campaignBoxGUID`,  However, if there is already a value set in that variable, it means more than one has been detected, so the error on 187 is thrown, and the process is aborted.

Pretty sure this was already functioning correctly with my previous push?  Regardless, I still think it is an extreme edge case; it would require a user manually deleting all the scenario boxes instead of recalling (since cleanup helper doesn't touch scenario boxes), and then placing a new campaign.  Still, I agree it's good to account for it and not leave it ambiguous. Makes the code neater and more predictable.